bot/cogs/bot.py

```diff
@@ -78,32 +80,35 @@ def __init__(self, bot: AutoShardedBot):
             if time.time()-self.previous_format_times[msg.channel.id] > 300 or msg.channel.id == DEVTEST_CHANNEL:
                 if msg.content.count("\n") >= 3:
                     try:
-                        # Some users formatted multi-line code using `'sip("
-                        # This fixes that by treating them as nothing.
-                        if "`" in msg.content:
-                            msg.content = msg.content[:-1][1:]
-                        tree = ast.parse(msg.content)
+                        # Filtering valid Python codeblocks and exiting if a valid Python codeblock is found
+                        if re.match(msg.content, "```(python|py)\n((?:.*\n*)+)```", re.IGNORECASE):
+                            return
+                        else:
+                            # Format the block into python source
+                            content = ""
+                            for line in msg.content.split("\n"):
+                                content += line.strip("`") + "\n"
+                            # Remove "Python" or "Py" from top of the message if exists
```

Comment:
blocks like this one should have a blank line above them, or the file starts to feel too dense. it's like a book without paragraphs, you know?